main.js

```diff
@@ -0,0 +1,305 @@
+  "03n":	"clouds-night",
+  "04n":	"clouds-night",
+  "09n":	"rain-night",
+  "10n":	"rain-night",
+  "11n":	"thunder-night",
+  "13n":	"snow-night",
+  "50n":	"mist-night"
+};
+
+//listening for click on search button, passing input into api fetch
+$('#search-button').click(function () {
+  var cityName = $('.city-name-input').val();
+  if (cityName === null) {
+    alert('Enter a city name');
+  };
+  daysOfWeekData = [[],[],[],[],[]];
+  forecastDataComputed = [[],[],[],[],[]];
+  fetch(cityName);
+  fetchForecast(cityName);
+  currentCityName = cityName;
+  //after a search, display 'set as default' button
+  $('.set-as-default').attr('id', 'display-button') 
+});
+//listen for click on set as defualt, if not "" put in session storage
+$('.set-as-default').click(function () {
+  if (currentCityName === "") {
+    console.log('in the if')
+    var cityName = $('.city-name-input').val()
+  } else {
+    cityName = currentCityName;
+  };
+  sessionStorage.setItem('userDefault', cityName);
+  alert('You set ' + cityName + ' as the default.');
+});
+//fetching current weather data
+var fetch = function (query) {
+$.ajax({
+  url: "http://api.openweathermap.org/data/2.5/weather?q=" + query + "&units=imperial&APPID=df3d2c2d8b73b0874f49b71164c4dcba",
+  type: "GET",
+  dataType: "jsonp",
+  success: function(data){
+      addWeather(data);
+      currentCityCoords.lat = data.coord.lat;
+      currentCityCoords.lon = data.coord.lon;
+      $('#map').removeAttr('class'); 
+      initMap();
+
+  },
+  error: function () {
+    alert('Bad input. Try again');
+  }
+});
+};
+//fetching 5 day forecast data
+var fetchForecast = function (query) {
+  $.ajax({
+    url: "http://api.openweathermap.org/data/2.5/forecast?q=" + query + "&units=imperial&APPID=df3d2c2d8b73b0874f49b71164c4dcba",
+    type: "GET",
+    dataType: "jsonp",
+    success: function(data){
+        addForecast(data)
+    },
+    error: function () {
+      alert('Bad input. Try again');
+    }
+  });
+  };
+//puts current weather api data into an obj, push to array
+var addWeather = function (data) {
+  weatherToday = [];
+  var conditionLink = data.weather[0].icon;
+  //checks the icon to set corresponding background color
+  setBackgroundColor(conditionLink);
+  var roundedTemp = Math.round(data.main.temp) + "\xB0";
+  var conditionLinkURL = 'http://openweathermap.org/img/wn/' + conditionLink + '@2x.png';
+  var newCity = {
+    currentCity: data.name || null,
+    currentTemp: roundedTemp || null,
+    currentCondition: data.weather[0].main || null,
+    imageURL: conditionLinkURL || null
+  }
+  weatherToday.push(newCity);
+  renderPage(); 
+};
+//puts 5 day forecast data into an obj, push to array
+var addForecast = function (data) {
+  //take api data, split up into dif arrays for each day of week
+  forecastDataToArrays(data);
+  getDayOfWeek();
+  getEachDayTemp();
+  getEachDayCondition();
+  getEachDayConditionIcon();
+  forecast = [];
+  var newForecast = {
```

Comment:
Instead of making this object so big, couldn't you just do an iteration through an array of 5 objects and create a new instance of your handlebars template through each iteration?